commit 44987f59e8aa873338071637dea24ef125e329a9
Author: Ian Tessier <ian@cim.mcgill.ca>
Date:   Wed Jul 25 12:22:35 2012 -0400

    The TIMER_PERIOD should be calculated using the timer interrupt frequency.

Index: linux-3.2.46/arch/arm/mach-bcm2708/bcm2708.c
===================================================================
--- linux-3.2.46.orig/arch/arm/mach-bcm2708/bcm2708.c	2013-07-26 19:34:01.000000000 +0000
+++ linux-3.2.46/arch/arm/mach-bcm2708/bcm2708.c	2013-07-26 19:34:10.000000000 +0000
@@ -618,7 +618,7 @@
 #endif
 }
 
-#define TIMER_PERIOD 10000	/* HZ in microsecs */
+#define TIMER_PERIOD DIV_ROUND_CLOSEST(STC_FREQ_HZ, HZ)
 
 static void timer_set_mode(enum clock_event_mode mode,
 			   struct clock_event_device *clk)
